docs/update_readme.py

```diff
@@ -0,0 +1,20 @@
+# Script to update end-to-end example in README.md.
+
+updated_readme = []
+with open('README.md', encoding="utf-8") as f:
+    in_stub = False
+    readme = f.readlines()
+    for line in readme:
+        if not in_stub:
+            updated_readme.append(line)
+        if line == "```python update-readme\n":
+            in_stub = True
+            with open('docs/tutorial/examples/hardmax_end_to_end.py', encoding="utf-8") as example_f:
+                example_code = example_f.readlines()
+                updated_readme += example_code
+        if line == "```\n" and in_stub:
```

Comment:
Maybe more robust to have some clear end-marker (rather than an empty-line) ... something that will NOT be in the replacement code ... so that running this update twice has the same behavior as running it once

Reply:
I guess a code block is of the format,
``` python code-block-name
code_stub
```
And the update_readme script only updates the code_stub section.

Reply:
Sorry, this looks fine. Agree that three backticks should be fine, I misread it.